### ISSUE TITLE
Renamed branch keeps incrementing pending changes as if they were not pushed to remote

### ISSUE THREAD

**Issue by maksimovic** [ForkIssues/Tracker#229]
Steps to reproduce:

1. Create a branch, and push it
2. Rename it and push it
3. Make a change, commit & push, and `(1)` will stay next to its name

The counter will keep incrementing when step 3 is repeated, although really there are no changes which haven't been pushed to the remote.

I think the solution which fixed things was to delete the first branch from the remote + fetch with "prune remote branches".

As it can be seen at the screenshot, `test-branch-renamed` has 2 commits which are pushed to the remote, but the counter next to its name shows `(2)`, most likely counting the number of commits related to the first branch which has been renamed and "stale", instead of relating the number of commits to the branch which the first one has been renamed to.

Sorry for the heavy blur in the screenshot 😬 

![test-branch-renamed_](https://user-images.githubusercontent.com/1126226/36719456-61494286-1ba5-11e8-981c-afff0ace3e5e.jpg)

**Comment by DanPristupov** [comment]
Pending changes should work correctly, it looks like there's a bug in push. It should NOT create a new remote branch after rename.

The only relation between local and remote branches is a tracking reference.

Here's the correct behavior:
```
1. create local branch A
A

2. push A to origin:
A -> origin/A (created origin/A and added tracked reference A -> origin/A)

3. rename local branch A
A1 -> origin/A (A1 still tracks origin/A)

4. push A1 to origin
A1 -> origin/A (A1 still tracks origin/A)
```

It looks like there's a bug in step 4. I'll investigate that.

Please check the tracking reference for local branch item tooltip on the sidebar. It should be the same as before renaming, i.e. `origin/testbranch`.

**Comment by maksimovic** [comment]
The renamed `A1` branch continues to track `origin/A` after the branch is renamed `A` -> `A1` and pushed.

**Comment by DanPristupov** [comment]
I confirm, this is a bug. Fork pushes to incorrect branch when local branch name and upstream differs. Will fix in the next update.

**Comment by DanPristupov** [comment]
@maksimovic Just reporting the state. I investigated the problem and turned out that you are right 👍. Behind/ahead information is not calculated for branches with different names.
Unfortunately I could not fix this quickly.

**Comment by DanPristupov** [comment]
> Behind/ahead information is not calculated for branches with different names.

I fixed that in 1.0.72
